src/plugins/syscalls/syscalls.cpp

```diff
@@ -128,10 +130,103 @@ static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
     return 0;
 }
 
+static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j,nargs;
+    unsigned char* buf; // pointer to buffer to hold argument values
+
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t*)info->trap->data;
+    syscalls *s = wrapper->sc;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    if(wrapper->syscall_index>-1) { // get arguments only if we know how many to get
+        nargs = win_syscall_struct[wrapper->syscall_index].num_args;
+
+        unsigned long size = s->reg_size * nargs;
+        buf = (unsigned char *)g_malloc(sizeof(char)*size);
```

Comment:
IMHO it would be a lot simpler if we just allocated `nargs*sizeof(addr_t)` space here, then read each argument into its respective location using `vmi_read_32` and `vmi_read_64`.